src/backend/src/services/reimbursement-requests.services.ts

```diff
@@ -445,7 +446,16 @@ export default class ReimbursementRequestService {
    * @returns the created vendor
    */
   static async createVendor(submitter: User, name: string) {
-    if (!isAdmin(submitter.role)) throw new AccessDeniedAdminOnlyException('create vendors');
+    const validateCreateVendorSubmitter = async (submitter: User): Promise<void> => {
+      const failedAuthorizationException = new AccessDeniedException(
+        'Only admins, finance leads, and finance heads can create vendors.'
+      );
+
+      const authorized = isAdmin(submitter.role) || (await isUserLeadOrHeadOfFinanceTeam(submitter));
```

Comment:
Can we rename this boolean variable to `isAuthorized`? Wanted to align with other boolean function variable names such as `isAdmin`, `isUserLeadOrHeadOfFinanceTeam`, and so on.